mongodb-setup/app/server/session.server.ts

```diff
@@ -0,0 +1,79 @@
+import { createCookieSessionStorage, redirect } from '@remix-run/node';
+import bcrypt from 'bcryptjs';
+import * as dotenv from 'dotenv';
+
+import type { LoginForm } from '~/types';
+import { UserApi } from '~/user/api';
+dotenv.config();
+
+const sessionSecret = process.env.COOKIE_SECRET;
+const userService = new UserApi();
+
+if (!sessionSecret) {
+    throw new Error('COOKIE_SECRET must be set');
+}
+
+const storage = createCookieSessionStorage({
+    cookie: {
+        name: process.env.COOKIE_NAME,
+        secure: process.env.NODE_ENV === 'production',
+        secrets: [sessionSecret],
+        sameSite: 'lax',
+        path: '/',
+        maxAge: 60 * 60 * 24 * 30,
+        httpOnly: true
+    }
+});
+export async function createUserSession(id: string, email: string, redirectTo: string) {
+    const session = await storage.getSession();
+    session.set('id', id);
+    session.set('email', email);
+    return redirect(redirectTo, {
+        headers: {
+            'Set-Cookie': await storage.commitSession(session)
+        }
+    });
+}
+function getUserSession(request: Request) {
```

Comment:
Let's make it more clear that this is an async function

```suggestion
async function getUserSession(request: Request) {
```